File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cs
public class Solution {
    public int LongestOnes(int[] nums, int k) {
        int n = nums.Length;
        int maxLen=0;
        for(int i=0;i<n;i++){
            int zeroes=0;
            for(int j=i;j<n;j++){
                if(nums[j]==0) zeroes++;
                    if(zeroes<=k){
                    int len=j-i+1;
                    maxLen = Math.Max(len, maxLen);
                    }else break;

                
            }
        }
        return maxLen;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int l=0,r=0,maxLen=0,zeroes=0,n=nums.length;
        while(r<n){
            if(nums[r]==0)zeroes++;
            if(zeroes>k){
                if(nums[l]==0){
                    zeroes--;
                }
                l++;
            }
            
            if(k>=zeroes){
                int len=r-l+1;
                maxLen=Math.max(len,maxLen);
            }
            r++;
        }
        return maxLen;
    }
}